pynest/examples/hpc_benchmark.py

```diff
@@ -0,0 +1,426 @@
+#     a = tau_m / tau_syn
+#     b = 1.0 / tau_syn - 1.0 / tau_m
+
+# Inverting this equation numerically leads to tau_syn = 0.32582722403722841 ms,
+# as specified in model_params below.
+
+tau_syn = 0.32582722403722841
+
+# -------------------------------------------------------------------------------------
+
+brunel_params = {
+    'NE': int(9000 * params['scale']),  # number of excitatory neurons
+    'NI': int(2250 * params['scale']),  # number of inhibitory neurons
+
+    'Nrec': 100,  # number of neurons to record spikes from
+
+    'model_params': {  # Set variables for iaf_psc_alpha
+        'E_L': 0.0,  # Resting membrane potential(mV)
+        'C_m': 250.0,  # Capacity of the membrane(pF)
+        'tau_m': 10.0,  # Membrane time constant(ms)
+        't_ref': 0.5,  # Duration of refractory period(ms)
+        'V_th': 20.0,  # Threshold(mV)
+        'V_reset': 0.0,  # Reset Potential(mV)
+        'tau_syn_ex': tau_syn,  # time const. postsynaptic excitatory currents(ms)
+        'tau_syn_in': tau_syn,  # time const. postsynaptic inhibitory currents(ms)
+        'tau_minus': 30.0,  # time constant for STDP(depression)
+        # V can be randomly initialized see below
+        'V_m': 5.7  # mean value of membrane potential
+    },
+
+    'randomize_Vm': True,
+    'mean_potential': 5.7,   # Note that Kunkel et al. (2014) report different values. The values
+    'sigma_potential': 7.2,  # in the paper were used for the benchmarks on K, the values given
+                             # here were used for the benchmark on JUQUEEN.
+
+    'delay': 1.5,  # synaptic delay, all connections(ms)
+
+    # synaptic weight
+    'JE': 0.14,  # peak of EPSP
+
+    'sigma_w': 3.47,  # standard dev. of E->E synapses(pA)
+    'g': -5.0,
+
+    'stdp_params': {
+        'delay': 1.5,
+        'alpha': 0.0513,
+        'lambda': 0.1,  # STDP step size
+        'mu': 0.4,  # STDP weight dependence exponent(potentiation)
+        'tau_plus': 15.0,  # time constant for potentiation
+    },
+
+    'eta': 1.685,  # scaling of external stimulus
+    'filestem': params['path_name']
+}
+
+'''
+ FUNCTION SECTION
+'''
+
+
+def build_network(logger):
+    tic = time.time()  # start timer on construction
+
+    # unpack a few variables for convenience
+    NE = brunel_params['NE']
+    NI = brunel_params['NI']
+    model_params = brunel_params['model_params']
+    stdp_params = brunel_params['stdp_params']
+
+    # set global kernel parameters
+    nest.SetKernelStatus({
+        'total_num_virtual_procs': params['nvp'],
+        'resolution': params['dt'],
+        'overwrite_files': True})
+
+    nest.SetDefaults('iaf_psc_alpha', brunel_params['model_params'])
+
+    nest.message(M_INFO, 'build_network', 'Creating excitatory population.')
+    E_neurons = nest.Create('iaf_psc_alpha', NE)
+
+    nest.message(M_INFO, 'build_network', 'Creating inhibitory population.')
+    I_neurons = nest.Create('iaf_psc_alpha', NI)  # subnet gets own gid
+
```

Comment:
What is the purpose of this comment? You are not creating a subnet here, are you?